src/app/components/Basic/intlHelpers/index.js

```diff
@@ -1,61 +1,46 @@
-  childContextTypes, getChildContext
-} from 'react-intl';
-
-const methods = [
-  'formatPlural',
-  'formatDate',
-  'formatTime',
-  'formatNumber',
-  'formatRelative'
-];
-
-export const intlMethods = (Component) => (
-
-  @injectIntl
-  class IntlComponent extends React.Component {
-
-    maybeWarnForOverride(method) {
-      if (process.env.NODE_ENV !== 'production') {
-        if (Component.prototype[method] && !Component.prototype[method].__mine) {
-          console.warn(`Overriding instance method '${method}' for Component '${Component.name}'`); // eslint-disable-line no-console
-        }
-      }
-    }
-
-    constructor(props) {
-      super(props);
-      // all normal intl methods
-      methods.forEach(method => {
-        if (!Component.prototype[method] || !Component.prototype[method].__mine) {
-          this.maybeWarnForOverride(method);
-
-          Component.prototype[method] = (...args) => {
-            return props.intl[method].call(this, ...args);
-          };
-          Component.prototype[method].__mine = true;
-        }
-      });
+  intl, FormattedDate, FormattedNumber, FormattedRelative, FormattedTime,
+  childContextTypes, getChildContext, contextTypes
+} from 'react-intl-hoc';
+
+import * as t from 'tcomb';
+import get from 'buildo-lodash-ts/lib/get';
+import {
+  keyMapFM,
+  intlMethods as originalIntlMethods
+} from 'react-intl-hoc';
+
+// TODO(gio): this is basically useless now
+// remove
+function morpheusForm(v) {
+  const msgs = this.context.messages;
+  if (!msgs || !t.Str.is(v)) {
+    return undefined;
+  }
+  const parts = t.Str.is(v) && v.split('.');
+  if (parts.length === 1) {
+    const [v0] = parts;
+    return v0 && msgs[v0] && v0;
+  } else if (parts.length === 2) {
+    const [v0, v1] = parts;
+    return msgs[`${v0}_fields`] && msgs[`${v0}_fields`][v1] && `${v0}_fields.${v1}`;
+  } else {
+    return undefined;
+  }
+}
 
-      // `formatMessage` passing flat arguments
-      if (!Component.prototype.formatMessage || !Component.prototype.formatMessage.__mine) {
-        this.maybeWarnForOverride('formatMessage');
+function morpheusMessageOrJustMessageOrNothing(msg) {
+  return morpheusForm.call(this, msg) || msg;
+}
 
-        Component.prototype.formatMessage = (key, ...args) => {
-          return props.intl.formatMessage.apply(this, [{ id: key }, ...args]);
-        };
-        Component.prototype.formatMessage.__mine = true;
-      }
-    }
+const translatableMatch = /^[A-Z][a-z]+(\.[a-z]+)+$/;
+function messageExists(key) {
+  return !!get(this.context.messages, key) ||
+    // throw for missing translation keys look alike
+    !!translatableMatch.exec(key);
+}
 
-    render() {
-      return <Component {...omit(this.props, 'intl')} />;
-    }
-  }
-);
+export const FormattedMessage = keyMapFM(morpheusMessageOrJustMessageOrNothing);
 
+export const intlMethods =
+  originalIntlMethods(morpheusMessageOrJustMessageOrNothing, messageExists);
```

Comment:
reminder for myself: this file can be simplified further (see alinity pro). Or we could even use a newer version of `react-intl` like we do in gdsm